### PR TITLE
unleavened critnib

### PR DESCRIPTION
**Pull request by kilobyte**
There's no need to for explicit leaves — the key is a part of the entry (value) struct.

---
This change is [Reviewable](https://reviewable.io/reviews/pmem/vmemcache/97)

### CODE REVIEW

File: src/critnib.c [hunk 1]
```diff
@@ -55,6 +55,8 @@
 #define NIB ((1 << SLICE) - 1)
 #define SLNODES (1 << SLICE)
 
+#define KEYLEN(l) (l->key.ksize + sizeof(size_t))
```
Comment (pbalcer) [inline src/critnib.c]: don't call variables `l`. It's in `CODING_STYLE.md` :)

**Comment by kilobyte** [comment]
That's possible in vmemcache (where we have struct cache_entry) but not in the general case.  This commit relies on knowing insides of the objects stored.

File: src/critnib.c (hunk 1 above)
Comment (kilobyte) [reply src/critnib.c]: Changed to "leaf" rather than "x" to make it self-documenting.

**Comment by ldorau** [comment]
This patch possibly fixes a memory leak detected by https://github.com/pmem/vmemcache/pull/93 and should be merged before https://github.com/pmem/vmemcache/pull/93

**Pull request merged**